src/shims/windows/foreign_items.rs

```diff
@@ -303,38 +395,73 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
                 this.write_scalar(handle.to_scalar(this), dest)?;
             }
             "GetFileInformationByHandle" => {
-                let [handle, info] = this.check_shim_sig_lenient(abi, sys_conv, link_name, args)?;
+                let [handle, info] = this.check_shim_sig(
+                    shim_sig!(extern "system" fn(winapi::HANDLE, *mut _) -> winapi::BOOL),
+                    link_name,
+                    abi,
+                    args,
+                )?;
                 let res = this.GetFileInformationByHandle(handle, info)?;
                 this.write_scalar(res, dest)?;
             }
             "SetFileInformationByHandle" => {
-                let [handle, class, info, size] =
-                    this.check_shim_sig_lenient(abi, sys_conv, link_name, args)?;
+                let [handle, class, info, size] = this.check_shim_sig(
+                    shim_sig!(
+                        extern "system" fn(
+                            winapi::HANDLE,
+                            winapi::FILE_INFO_BY_HANDLE_CLASS,
+                            *mut _,
+                            u32,
+                        ) -> winapi::BOOL
+                    ),
+                    link_name,
+                    abi,
+                    args,
+                )?;
                 let res = this.SetFileInformationByHandle(handle, class, info, size)?;
                 this.write_scalar(res, dest)?;
             }
             "FlushFileBuffers" => {
-                let [handle] = this.check_shim_sig_lenient(abi, sys_conv, link_name, args)?;
+                let [handle] = this.check_shim_sig(
+                    shim_sig!(extern "system" fn(winapi::HANDLE) -> winapi::BOOL),
+                    link_name,
+                    abi,
+                    args,
+                )?;
                 let res = this.FlushFileBuffers(handle)?;
                 this.write_scalar(res, dest)?;
             }
             "DeleteFileW" => {
-                let [file_name] = this.check_shim_sig_lenient(abi, sys_conv, link_name, args)?;
+                let [file_name] = this.check_shim_sig(
+                    shim_sig!(extern "system" fn(*const _) -> winapi::BOOL),
+                    link_name,
+                    abi,
+                    args,
+                )?;
                 let res = this.DeleteFileW(file_name)?;
                 this.write_scalar(res, dest)?;
             }
             "SetFilePointerEx" => {
-                let [file, distance_to_move, new_file_pointer, move_method] =
-                    this.check_shim_sig_lenient(abi, sys_conv, link_name, args)?;
+                let [file, distance_to_move, new_file_pointer, move_method] = this.check_shim_sig(
+                    // i64 is actually a LARGE_INTEGER union of {u32, i32} and {i64}
```

Comment:
And that becomes just `i64` on the ABI level? Those types are not ABI-compatible under Rust rules... but I guess they are compatible on Windows targets so "it's fine".^^